src/ResultSet.php

```diff
@@ -2,38 +2,46 @@
 
 namespace Yiisoft\Validator;
 
+/**
+ * ResultSet stores validation result of each attribute from {@link DataSetInterface}.
+ * It is typically obtained by validating data set with {@link Validator}.
+ */
 final class ResultSet implements \IteratorAggregate
 {
     /**
      * @var Result[]
      */
-    private $results = [];
+    private array $results = [];
 
-    public function addResult(string $attribute, Result $result): void
+    public function addResult(string $attribute, Result $result): self
```

Comment:
`addResult(Result $result, ?string $attribute = null):self`
- For a primitive type result, the attribute is irrelevant.
- For a dataset type result, the attribute is require

Reply:
ResultSet is for data set validation results, not primitive results.